Allow extension authors to set valueSelection in a QuickPick/InputBox

r3m0t opened the pull request:
<!-- Thank you for submitting a Pull Request. Please:
* Read our Pull Request guidelines:
  https://github.com/microsoft/vscode/wiki/How-to-Contribute#pull-requests
* Associate an issue with the Pull Request.
* Ensure that the code is up-to-date with the `main` branch.
* Include a description of the proposed changes and how to test them.
-->

Similar to #119144 @TylerLeonhardt 

Allows `showQuickPick` and `showInputBox` to be used with `step, totalSteps` so that extension authors don't need to use the more complex `createQuickPick` and `createInputBox`.

Also allow `valueSelection` to be set with `createQuickPick` and `createInputBox`.

Test with this extension- https://github.com/r3m0t/vscode-extension-samples/tree/quickinput/quickinput-sample

TylerLeonhardt commented:
I'll bring this to the API review but I won't get hopes too high. We intentionally haven't done this in the past is because the `pick` API is suppose to be for one-offs picks while the `createQuickPick` is for longer form quick pick use cases (multi-step).

With this proposal, you'll have to keep track of `totalSteps` outside of the quickpick and pass that into `pick` every single time.

I do acknowledge that the `pick` API is simpler than `createQuickPick` and this PR enables:

```ts

const a = await window.showInputBox({
		title: "Example",
		prompt: "Please select step 1's value",
		step: 1,
		totalSteps: 3,
	}, ['a', 'b', 'c']);

const b = await window.showInputBox({
		title: "Example",
		prompt: "Please select step 2's value",
		step: 2,
		totalSteps: 3,
	}, ['a', 'b', 'c']);
```

but that's actually 2 pieces data that are duplicated: `title` and `totalSteps`... which isn't ideal and would be avoided using `createQuickPick`.

r3m0t commented:
I'm more interested in valueSelection actually,  which I implemented in this PR. I am using the multi step input and want to set valueSelection on the second step. I just added step to QuickPickOptions for symmetry.

TylerLeonhardt commented:
Yes that's right, I remember your issue. I'd rather have `valueSelection?: [number, number]` on `InputBox` then `totalSteps` on `pick`.

TylerLeonhardt commented:
Hi @r3m0t we discussed this in our API review because I wanted to get you an answer for taking this on.

We all feel similar to what I outlined above with the steps, but would 💯 take a PR with the `valueSelection`. If you update this PR with that, I'd happily get that in!

r3m0t commented:
@TylerLeonhardt there you go, I kept the other branch in `r3m0t/quick-pick-steps-c`

TylerLeonhardt commented on src/vscode-dts/vscode.d.ts:
```diff
@@ -2018,10 +2018,10 @@ declare module 'vscode' {
 		 * otherwise the defined range will be selected.
 		 */
-		valueSelection?: [number, number];
+		valueSelection?: readonly [number, number];
```
This is a slightly breaking change so please leave this as it is for now.

r3m0t commented:
@TylerLeonhardt all sorted

jrieken commented on src/vs/workbench/browser/actions/workspaceCommands.ts:
```diff
@@ -235,7 +235,7 @@ CommandsRegistry.registerCommand({
 	},
 	description: {
-		description: 'Opens an new window depending on the newWindow argument.',
+		description: 'Opens an new window depending on the reuseWindow argument.',
```
unrelated change sin this file?

r3m0t replied:
It is an accurate change to the description, but you can remove it if you like

r3m0t replied:
@jrieken removed

TylerLeonhardt replied:
In this case, it's ok. I've already reviewed all the changes but @jrieken brings up a really good point for future PRs.
